spec/headers_spec.rb

```diff
@@ -35,13 +35,39 @@
         let(:uri) { 'http://google.com/?redirect_to=https://www.example.com'.freeze }
 
         it 'return /?redirect_to=https://www.example.com as canonical string path' do
-          expect(subject.canonical_string).to eq('GET,,,/?redirect_to=https://www.example.com,')
+          expect(subject.canonical_string).to eq('GET,,,/?redirect_to=https%3A%2F%2Fwww.example.com,')
         end
 
         it 'does not change request url (by removing host)' do
           expect(request.url).to eq(uri)
         end
       end
+
+      if RUBY_VERSION.to_f > 2.1
```

Comment:
Why do you need this ?

Reply:
The rest-client will throw an exception for an invalid URI in ruby versions 2.1 and below, I needed this so that all the CI tests would pass.  I believe, in later versions the URI is properly escaped rather than throwing the exception but I did not verify in the rest-client code.